### PR TITLE
Solution for Issue #19 (implemented function eport_event_agenda(...)

### PR DESCRIPTION
**Pull request by kolibri52**

### CODE REVIEW

File: ChurchToolsApi.py [hunk 1]
```diff
@@ -538,6 +538,41 @@ def get_event_agenda(self, event_id: int):
...
+        :return: bool if success
+        """
+        url = '{}/api/agendas/{}/export'.format(self.domain, agenda_id)
+        # NOTE the stream=True parameter below
+        params = {
+            'target': target_format
+        }
+        # The following 3 parameter are mandatory from the churchtools API side:
+        json_data = {
+            'appendArrangement': True,
```
Comment (bensteUEM) [inline ChurchToolsApi.py:557]: Isn't this a custom param which should be part of the method arguments?
e.g. we always download songs without appending Arrangement ... or is this a different option?

File: ChurchToolsApi.py [hunk 2]
```diff
@@ -538,6 +538,41 @@ def get_event_agenda(self, event_id: int):
...
+            'exportSongs': True,
+            'withCategory': True,
+        }
+        headers = {
+            'accept': 'application/json',
+            'Content-Type': 'application/json',
+        }
+
+        response = self.session.post(url=url, params=params, headers=headers, json=json_data)
+        ResultOK = False
```
Comment (bensteUEM) [inline ChurchToolsApi.py:567]: variable names should start lower case
please rename to either result_ok or resultOk

File: ChurchToolsApi.py [hunk 3]
```diff
@@ -538,6 +538,41 @@ def get_event_agenda(self, event_id: int):
...
+        if response.status_code == 200:
+            response_content = json.loads(response.content)
+            agenda_data = response_content['data'].copy()
+            logging.debug("Agenda package found {}".format(response_content))
+            ResultOK = self.file_download_from_url('{}/{}'.format(self.domain, agenda_data['url']), target_path)
+
```
Comment (bensteUEM) [inline ChurchToolsApi.py:573]: please include ELSE with error or warning being logged similar to other methods
e.g. 
    ```
    else:
            logging.warning("Something went exporting event_agenda: {}".format(response.status_code))
```

File: ChurchToolsApi.py [hunk 4]
```diff
@@ -633,4 +668,4 @@ def file_download_from_url(self, file_url: str, target_path: str):
                 return True
             else:
                 logging.warning("Something went wrong during file_download: {}".format(r.status_code))
-                return False
\ No newline at end of file
+                return False
```
Comment (bensteUEM) [inline ChurchToolsApi.py:671]: This is not related to the commit topic and should usually not be part of the commit - i suppose you can't exclude it anymore?

File: ChurchToolsApi.py [hunk 5]
```diff
@@ -538,6 +538,41 @@ def get_event_agenda(self, event_id: int):
             logging.info("Event requested that does not have an agenda with status: {}".format(response.status_code))
             return None
 
+    def export_event_agenda(self, target_format, agenda_id, target_path):
```
Comment (bensteUEM) [inline ChurchToolsApi.py:541]: please set a default for target path e.g. 
`path_for_download='./Downloads' `
similar to file_download

File: ChurchToolsApi.py (hunk 3 above)
Comment (bensteUEM) [inline ChurchToolsApi.py:573]: What do you think about adding 
logging.debug('download finished')
or similar at this point?

**Review by bensteUEM — COMMENTED** [review]
Please have a look at the review lines - most of them are minor changes requests. Would be great if you would consider them.

File: ChurchToolsApi.py (hunk 1 above)
Comment (kolibri52) [reply ChurchToolsApi.py:557]: True means the name of the arrangement will be included within the agenda caption - for example in Songbeamer.
Okay I will add it to  the method arguments.

File: ChurchToolsApi.py (hunk 3 above)
Comment (kolibri52) [reply ChurchToolsApi.py:573]: Okay. Good idea -> Done

File: ChurchToolsApi.py (hunk 5 above)
Comment (kolibri52) [reply ChurchToolsApi.py:541]: I've added a default path. Moreover it is possible to define the exact filename or only a foldername (then Defaultfilename is used) for the download now.

File: ChurchToolsApi.py (hunk 3 above)
Comment (kolibri52) [reply ChurchToolsApi.py:573]: Good idea. Done.

**Comment by kolibri52** [comment]
I will create a new pull request which targets the new feature branch,
in order to not include unnecessary changes.
